### PR TITLE
(test) O3-5608: Add unit tests for convertTime12to24 time helper

### PR DESCRIPTION
**Pull request by sohamdhande**
## Requirements
- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a conventional commit label. See existing PR titles for inspiration.
- [ ] My work is based on designs, which are linked or shown either in the Jira ticket or the description below.
- [x] My work includes tests or is validated by existing tests.

## Summary
Adds unit tests for the `convertTime12to24` utility function in `helpers/time.ts`.

Tests cover:
- AM conversion (including 12:00 AM midnight edge case)
- PM conversion (including 12:00 PM noon edge case)
- All valid AM/PM hours (1-11)
- Malformed input (no colon, empty string, multiple colons, whitespace)
- Return value structure (array length, type of hours for AM vs PM)

These tests help coding agents verify their work in a feedback loop when modifying time-related logic.

## Screenshots
N/A — no UI changes.

## Related Issue
N/A

## Other
N/A

### CODE REVIEW

File: packages/esm-appointments-app/src/helpers/__tests__/time.test.ts [hunk 1]
```diff
@@ -0,0 +1,233 @@
+import { convertTime12to24, amPm } from '../time';
```
Comment (jayasanka-sack) [inline packages/esm-appointments-app/src/helpers/__tests__/time.test.ts]:
```suggestion
import { convertTime12to24 } from '../time';
```

Seems this is an unused import.

File: packages/esm-appointments-app/src/helpers/__tests__/time.test.ts [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+    });
+
+    it('should convert 1:15 PM to 13:15', () => {
+      const result = convertTime12to24('1:15', 'PM');
+      expect(result).toEqual([13, '15']);
+    });
+
+    it('should convert 11:59 PM to 23:59', () => {
+      const result = convertTime12to24('11:59', 'PM');
+      expect(result).toEqual([23, '59']);
```
Comment (jayasanka-sack) [inline packages/esm-appointments-app/src/helpers/__tests__/time.test.ts:69]: I was wondering why some values were numbers and others were strings, then realized anything over 12 turns into an int because of this line:

```
hours = hours === '12' ? hours : parseInt(hours, 10) + 12;
```

I think it should always return strings instead. Can you update the `convertTime12to24` function as a follow-up and make sure it uses proper types instead of `any`?

**Review by jayasanka-sack — APPROVED** [review]
Thanks a lot for getting this done @sohamdhande. Very smart!

**Comment by sohamdhande** [comment]
Done @jayasanka-sack, updated the test expectations to match the string return type. Also updated the return value structure test description to reflect the new consistent behaviour.

**Comment by jayasanka-sack** [comment]
@sohamdhande seems like you misunderstood what I've mentioned. You shouldn't change anything in tests, instead fix the `convertTime12to24` function to always return strings for hours. You can do it as a followup PR. 

Please undo https://github.com/openmrs/openmrs-esm-patient-management/pull/2489/changes/d920bca9c09b1fd4f9a791312688b87db8edf447. What you had was correct.

**Review by jayasanka-sack — CHANGES_REQUESTED** [review]
Please revert the last commit.

**Comment by sohamdhande** [comment]
@jayasanka-sack. My bad, I misread your comment as updating the tests to match the current behavior, when you actually wanted the function fixed to match what the tests already expressed. Reverted d920bca. Will raise a follow-up PR fixing convertTime12to24 to always return strings and update the types accordingly.

**Comment by jayasanka-sack** [comment]
Thanks @sohamdhande!

Merging this PR as it has nothing to do with E2E failures introduced by https://github.com/openmrs/openmrs-esm-patient-management/commit/da76c66478bfd976b822926cb43f2df97bf09005
